Util/InformationTheory/src/main/java/org/tribuo/util/infotheory/InformationTheory.java

```diff
@@ -136,10 +135,25 @@ public static <T1,T2,T3> GTestStatistics gTest(List<T1> first, List<T2> second,
             tuple = innerConditionalMI(first,second,conditionList);
         }
         double gMetric = 2 * second.size() * tuple.score;
-        ChiSquaredDistribution dist = new ChiSquaredDistribution(tuple.stateCount);
-        double prob = dist.cumulativeProbability(gMetric);
-        GTestStatistics test = new GTestStatistics(gMetric,tuple.stateCount,prob);
-        return test;
+        double prob = computeChiSquaredProbability(tuple.stateCount, gMetric);
+        return new GTestStatistics(gMetric,tuple.stateCount,prob);
+    }
+
+    /**
+     * Computes the cumulative probability of the input value under a Chi-Squared distribution
+     * with the specified degrees of Freedom.
+     * @param degreesOfFreedom The degrees of freedom in the distribution.
+     * @param value The observed value.
+     * @return The cumulative probability of the observed value.
+     */
+    private static double computeChiSquaredProbability(int degreesOfFreedom, double value) {
```

Comment:
FYI - this method does not work for odd values of `degreesOfFreedom`.  

`from scipy.stats import chi2`
`print(chi2.cdf(3.84,3))`

maybe that's why this method is private....

Reply:
works fine for odd values, btw.

`@Test`
`void testComputeChiSquaredProbability() throws Exception {`
`//        assertEquals(0.9499564787512949, InformationTheory.computeChiSquaredProbability(1, 3.84), 1e-14);`
`        assertEquals(0.8533930378696499, InformationTheory.computeChiSquaredProbability(2, 3.84), 1e-14);`
`//        assertEquals(0.7207323828813903, InformationTheory.computeChiSquaredProbability(3, 3.84), 1e-14);`
`        assertEquals(0.5719076705793775, InformationTheory.computeChiSquaredProbability(4, 3.84), 1e-14);`
`        assertEquals(0.9995692574594243, InformationTheory.computeChiSquaredProbability(2, 15.5), 1e-14);`
`    }`

Reply:
Yeah, I need it to compute the GTest but don't want to expose it out as I've not checked that it's valid for other use cases. I'd like to add a fuller stats package to Tribuo at some point, but in the meantime these functions will just appear as private things that we need for various hypothesis tests that we plan to add.